subsys/pm/device_runtime.c

```diff
@@ -514,17 +548,17 @@ int pm_device_runtime_disable(const struct device *dev)
 
 	if (!k_is_pre_kernel()) {
 		if ((pm->base.state == PM_DEVICE_STATE_SUSPENDING) &&
-			((k_work_cancel_delayable(&pm->work) & K_WORK_RUNNING) == 0)) {
+			((k_work_cancel_delayable(&pm->base.work) & K_WORK_RUNNING) == 0)) {
 			pm->base.state = PM_DEVICE_STATE_ACTIVE;
 			goto clear_bit;
 		}
 
 		/* wait until possible async suspend is completed */
 		while (pm->base.state == PM_DEVICE_STATE_SUSPENDING) {
-			k_event_clear(&pm->event, EVENT_MASK);
+			k_event_clear(&pm->base.event, EVENT_MASK);
```

Comment:
maybe `k_work_flush` could be used here so that event is not needed?
Also it would be good to add an assert here that checks if `pm_device_runtime_disable` is not called from the system work queue.

Reply:
Maybe, but not related to this PR.

Reply:
Related in a way that now event is added to base and increases RAM usage for ISR_SAFE case.